### PR TITLE
Fixes the scrolling sleep after click

### PR DESCRIPTION
**Pull request by gcandal**
Sleeps after scrolling at the interactions level, rather than the driver level. The rationale is that this is something not specific to selenium, but rather an higher level operation.

This was causing a bug when used in Retail because it tried to scroll twice in a row and the scroll there is not atomic (in the sense that the second scroll does not wait for the first and will cause it to misbehave).

### CODE REVIEW

File: src/ripe_rainbow/interactive/drivers.py [hunk 1]
```diff
@@ -125,15 +125,14 @@ def write_text(self, element, text):
...
+    def click(self, element, scroll = True):
         from selenium.webdriver.common.action_chains import ActionChains
         from selenium.common.exceptions import ElementClickInterceptedException, ElementNotVisibleException, WebDriverException
 
         try:
             if scroll:
-                # runs the scroll operation with the request amount of sleep time
-                # for the scroll operation (important to guarantee visibility)
-                self.scroll_to(element, sleep = scroll_sleep)
+                # runs the scroll operation (important to guarantee visibility)
```
Comment (gcandal) [inline src/ripe_rainbow/interactive/drivers.py]: Tbh I wanted to remove this part too and just let the interactions perform the scroll. Shouldn't be a problem given this is an internal API, right?

In any case, I'm running the tests without this to see what happens.

Comment (gcandal) [reply src/ripe_rainbow/interactive/drivers.py]: Confirmed: this is not needed.

**Comment by joamag** [comment]
@gcandal let's talk face to face next Monday so that you can explain the rationale behind this fix

File: src/ripe_rainbow/domain/logic/ripe_retail.py [hunk 2]
```diff
@@ -112,6 +114,11 @@ def set_part(
             ".pickers .button-part > p:not(.no-part)",
             condition = lambda e: e.text == part.upper()
         )
+
+        # the color transition must finish before we can keep going,
+        # otherwise scrolling to the correct color won't work
+        time.sleep(1)
```
Comment (joamag) [inline src/ripe_rainbow/domain/logic/ripe_retail.py]: Why is this necessary? click_when_possible should guarantee click only when the swatch is visible

Comment (gcandal) [reply src/ripe_rainbow/domain/logic/ripe_retail.py]: Because "when possible" = "is visible" (display != none & opacity > 0), which in this case is not enough.

Comment (joamag) [reply src/ripe_rainbow/domain/logic/ripe_retail.py]: We must fix the base code then

Comment (gcandal) [reply src/ripe_rainbow/domain/logic/ripe_retail.py]: When I answered I thought this was about something else. The problem here is what says in the comment, without the sleep this might sometimes happen:
1. Click part
2. Retail starts scrolling the colors to show the right color for that part
3. Because Rainbow doesn't wait for step 2 to finish it tries to click a color
4. Clicking a color triggers the scrolling to that color
5. Somehow the two scrolls conflict and the prevailing one is from step 2, resulting in the click for color selection not working

**Comment by gcandal** [comment]
Closed in favor of https://github.com/ripe-tech/ripe-rainbow/pull/39
